src/mc_rbdyn/ForceSensor.cpp

```diff
@@ -28,21 +28,21 @@ struct ForceSensorCalibData
    * should contain 13 parameters in that order: mass (1),
    * rpy for X_f_ds (3), position for X_p_vb (3), wrench
    * offset (6).
-   * @return Default calibration if the file does not exist,
-   * throws if the file is ill-formed. */
-  ForceSensorCalibData(const std::string & filename, const Eigen::Vector3d & gravity) : ForceSensorCalibData()
+   *
+   * If the file does not exist, returns a default calibration. If the file
+   * exists but its parameters are invalid, an exception will be thrown.
+   *
+   * @return false if the file was not found, true otherwise
+   *
+   * @throws std::invalid_argument if the file is ill-formed.
+   **/
+  bool loadData(const std::string & filename, const Eigen::Vector3d & gravity)
   {
-    loadData(filename, gravity);
-  }
-
-  void loadData(const std::string & filename, const Eigen::Vector3d & gravity)
-  {
-    const int nr_params = 13;
+    constexpr int nr_params = 13;
     std::ifstream strm(filename);
     if(!strm.is_open())
     {
-      LOG_ERROR("Could not open " << filename)
```

Comment:
I think this should be kept, in addition with checking the file existence before trying to open it (the file could exist but not be accessible for some reason)

Reply:
I see, in that case I've added a check for file existance that displays a nice error message if the file does not exist, and attempts to load it otherwise. It'll throw if the file parameters are invalid.

> No calibration file /home/arnaud/src/catkin_data_ws/src/hrp2_drc/hrp2_drc_description/calib/hrp2_drc/calib_data.LeftFootForceSensor found for force sensor LeftFootForceSensor

If the parameters are invalid, the exception will be even more explicit:

> [ForceSensorCalibData] Invalid calibration file /home/arnaud/src/catkin_data_ws/src/hrp2_drc/hrp2_drc_description/calib/hrp2_drc/calib_data.LeftFootForceSensor. The file should contain 13 parameters in that order:
    - mass (1)
    - rpy of local rotation between the model force sensor and real one (3)
    - translation from the parent body to the virtual link CoM (3)
    - wrench offset (6).